### ISSUE TITLE
issue: Problem with constraining an optional field

### ISSUE THREAD

**Issue by DimRev** [react-hook-form/react-hook-form#12087]
### Version Number

7.52.1

### Codesandbox/Expo snack

https://codesandbox.io/p/devbox/email-optional-2p94xy?file=%2Fpackage.json

### Steps to reproduce

1. On the main page of the sendbox try filling the email / write anything
2. "regret the email" and choose to leave it blank i.e. remove the text (the field will be left with "")
3. Now you have an error of an optional field that requires an email | undefined and you got no way to remove it

### Expected behaviour

Set optional fields to an undefined state if they are an empty string, though this might trigger an error from react with the field changing types.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by DimRev** [comment]
Yup that works, but I see that as a hack to get over the problem that the original version of this isn't doing what it supposed to.

This also makes the schema code much more verbose and not immediately understandable if you are dealing with large forms where there are lots of optional social links/emails and so on.

**Comment by Cbridger-mywellabee** [comment]
Unsure why you're raising the issue here as this just executes the zod schema against the object you give it, e.g if you run this:
```ts
// default values you're passing in OR values when you delete back to an empty string
  console.log(formSchema.safeParse({ email: "", emailWithPreProcessing: "" })); 
```

you get:
```
{
    "success": false,
    "error": {
        "issues": [
            {
                "validation": "email",
                "code": "invalid_string",
                "message": "Invalid email",
                "path": [
                    "email"
                ]
            }
        ],
        "name": "ZodError"
    },
    "_error": {
        "issues": [
            {
                "validation": "email",
                "code": "invalid_string",
                "message": "Invalid email",
                "path": [
                    "email"
                ]
            }
        ],
        "name": "ZodError"
    }
}
```

which is nothing to do with react-hook-form, react hook form is just rendering the errors it's getting from zod when it passes the values in, and getting react hook form to manipulate the values to be something other than what they actually are is definitely not the solution.

**Comment by DimRev** [comment]
Gotcha, my bad I assumed that an empty input field should return `undefined` instead of `""`, but if that's the correct behavior then it's I just need to change my schema.

**Comment by jedwards1211** [comment]
I too would rather see an empty input field return `undefined` by default.  It's more likely that the user meant no value than specifically the empty string or some whitespace.
